include/cudaq/Frontend/nvqpp/ASTBridge.h

```diff
@@ -144,18 +144,19 @@ class QuakeBridgeVisitor
   using Base = clang::RecursiveASTVisitor<QuakeBridgeVisitor>;
 
 public:
-  explicit QuakeBridgeVisitor(clang::ASTContext *astCtx,
-                              mlir::MLIRContext *mlirCtx, mlir::OpBuilder &bldr,
-                              mlir::ModuleOp module, SymbolTable &symTab,
-                              EmittedFunctionsCollection &funcsToEmit,
-                              llvm::ArrayRef<clang::Decl *> reachableFuncs,
-                              MangledKernelNamesMap &namesMap,
-                              clang::CompilerInstance &ci,
-                              clang::ItaniumMangleContext *mangler)
+  explicit QuakeBridgeVisitor(
+      clang::ASTContext *astCtx, mlir::MLIRContext *mlirCtx,
+      mlir::OpBuilder &bldr, mlir::ModuleOp module, SymbolTable &symTab,
+      EmittedFunctionsCollection &funcsToEmit,
+      llvm::ArrayRef<clang::Decl *> reachableFuncs,
+      MangledKernelNamesMap &namesMap, clang::CompilerInstance &ci,
+      clang::ItaniumMangleContext *mangler,
+      llvm::SmallVector<llvm::StringRef> &customOperations)
```

Comment:
Since we are identifying custom operations by their extern "C" generator functions, and those functions are marked with a new function attributed defined in qubit_qis.h, we decided to follow the same pattern the bridge takes for finding QPU kernels. There we find the FunctionDecls marked with __qpu__ and add them to the list to be processed to MLIR. It is the QPUCodeFinder that does that, not the QuakeBridgeVisitor, so we have to send the data over to the QuakeBridgeVisitor. Then we have something to check for when we see an operation that is not in the default set. 

Definitely open to any simpler ways you can think of though.

Reply:
This was updated in the latest commits. We do use the additional argument for custom operations, but, we are now using a map of the generator function and its mangled name (no longer using `cudaq::unitary_operation` since it cannot be handled by kernel (yet).
The general comment to handle macro expansion separately from `__qpu__` still applies and needs refactoring which I will do as a follow-up.